pkg/fanal/secret/builtin-rules.go

```diff
@@ -270,12 +270,13 @@ var builtinRules = []Rule{
 		Keywords: []string{"p8e-"},
 	},
 	{
-		ID:       "alibaba-access-key-id",
-		Category: CategoryAlibaba,
-		Title:    "Alibaba AccessKey ID",
-		Severity: "HIGH",
-		Regex:    MustCompile(`(LTAI)(?i)[a-z0-9]{20}`),
-		Keywords: []string{"LTAI"},
+		ID:              "alibaba-access-key-id",
+		Category:        CategoryAlibaba,
+		Title:           "Alibaba AccessKey ID",
+		Severity:        "HIGH",
+		Regex:           MustCompile(`([^0-9a-z]|^)(?P<secret>(LTAI)(?i)[a-z0-9]{20})([^0-9a-z]|$)`),
```

Comment:
I see `^0-9a-z` is characters other than digits and alphabetic characters, but what does `|^` mean?

Reply:
`|^` is the start line.
if the file only contains a secret, Trivy will not find that secret.
for example: https://regex101.com/r/v9TsI0/1.

same for `|$`

Reply:
Oh, I got it. Thanks.